### DISCUSSION TITLE
Use with PL 1.2.3

### DISCUSSION THREAD

**Discussion by hadim** [Lightning-AI/torchmetrics#81] [Q&A] [answered]
Is that already possible to use the `metrics` package with PL1.2.3 or do we need to work on `master` or wait with next PL release?

**Comment by SkafteNicki** [comment]
It should already be possible, everything should work as you know it from `pytorch_lightning.metrics`.

**Reply by FlorianMF** [reply]
Using 
```
self.train_acc(pred,target)
self.log('train_acc', self.train_acc, ..., on_epoch=True)
```
does not work though, because "reduce_on_epoch_end" does check whether the output element is a pl.Metric. 
As the TorchMetrics instances aren't, the function tries to convert it to a tensor which leads to a RunTimeError.

**Reply by SkafteNicki** [reply]
Thanks @FlorianMF for correcting me.
When this PR https://github.com/PyTorchLightning/pytorch-lightning/pull/6540 gets merged on the lightning side it should work.

**Reply by SkafteNicki** [reply]
@FlorianMF yes, that is the idea. `pytorch_lightning.metrics` will still exist, but will be deprecated from v1.3 and removed in v1.5 of lightning. Also, all metrics in lightning will just (until removed) call torchmetrics underneath.

**Reply by Borda** [reply]
yes, https://github.com/PyTorchLightning/pytorch-lightning/pull/6540 and https://github.com/PyTorchLightning/pytorch-lightning/pull/6530 were merged 🐰

**Comment by Borda** [comment, marked as answer]
base on the comment above and https://github.com/PyTorchLightning/pytorch-lightning/pull/6540 and https://github.com/PyTorchLightning/pytorch-lightning/pull/6530 were merged 🐰  you shall be able to use it from master and atm we ain on full re-routing for PL 1.3 ⚡

**Reply by hadim** [reply]
Will we be able to use `torchmetrics` with 1.2.4 (or any releases before 1.3)?

**Reply by Borda** [reply]
v1.2.4 is out and it is not there, but if there is such hunger we can think about it for v1.2.5
